docs/build/message-history.md

```diff
@@ -0,0 +1,138 @@
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Enable message history for group chat with XMTP
+
+Each app installation generates unique group chat keys. This means that group chat messages sent from one app installation can only be decrypted by that same app installation.
+
+To enable users to access their group chat messages using all of their app installations, enable the message history feature. Enabling message history is necessary for a seamless multi-app installation experience.
+
+To learn more about the user experience without the message history feature, see [⚠️ Important: Manage actions that make a local database inaccessible](/docs/build/group-chat/#️-important-manage-actions-that-make-a-local-database-inaccessible).
+
+## Run a message history server
+
+To enable the message history feature, you must run your own [**xmtp-message-history-server**](https://github.com/xmtp/xmtp-message-history-server). When you do, it will provide a message history sync URL to use in the next step.
+
+## Set message history sync URL
+
+When your app creates a new client, specify the location where the client should sync message history. Whether you support the message history feature in your app or not, we recommend setting this value to allow other apps to pull in your app's message history. The message history starts at the point you set this value.
+
+<Tabs groupId="sdk-langs">
+<TabItem value="rn" label="React Native" attributes={{className: "rn_tab"}}>
+
+```jsx
+  const clientOptions = {
+    env: 'production' as const,
+    enableV3: true,
+    appVersion: 'YourApp/1.0.0',
+    dbEncryptionKey: encryptionKey, // Replace with actual Array encryption key
+    // Add the history sync URL option
+    historySyncUrl: 'SYNC_URL'
+  }
+```
+
+</TabItem>
+<TabItem value="kotlin" label="Kotlin" attributes={{className: "kotlin_tab"}}>
+
+```kotlin
+Client().create(
+    account = alixWallet,
+    options = ClientOptions(
+        ClientOptions.Api(XMTPEnvironment.PRODUCTION, true),
+        enableV3 = true,
+        dbEncryptionKey: encryptionKey, // Replace with actual ByteArray encryption key
+        appContext = context
+        historySyncUrl = "SYNC_URL"
+    )
+)
+```
+
+</TabItem>
+<TabItem value="swift" label="Swift"  attributes={{className: "swift_tab"}}>
+
+```swift
+let options = ClientOptions(
+    api: ClientOptions.Api(
+        env: .production,
+        isSecure: true
+    ),
+    enableV3: true,
+    dbEncryptionKey: encryptionKey, // Replace with actual String encryption key
+    historySyncUrl: "SYNC_URL"
+)
+```
+
+</TabItem>
+<TabItem value="node" label="Node"  attributes={{className: "node_tab"}}>
+
+```tsx
+// this API is experimental and may change in the future
+
+const client = await Client.create('0x1234...', {
+  apiUrl: 'SYNC_URL'
+});
+```
+
+</TabItem>
+</Tabs>
+
+## Request message history sync
+
+Code your app to request a message history sync anytime a user accesses a conversation view. 
```

Comment:
When we say "conversation view" - just to be sure - Is this the view when a user opens a group chat and can see the messages in the group chat?